File: CsharpProjects/myLeetCode/String/Offer 05. 替换空格.cs
// 剑指 Offer 05. 替换空格
// https://leetcode.cn/problems/ti-huan-kong-ge-lcof/
public class Solution {
    public string ReplaceSpace(string s) {
        return s.Replace(" ", "%20");
    }
}
public class Program {
    public static void Main() {
        Solution solution = new Solution();

        string input = "We are happy.";
        string result = solution.ReplaceSpace(input);

        Console.WriteLine($"Input: '{input}'");
        Console.WriteLine($"Output: '{result}'");
    }
}

File: C++/代码随想录/String/Offer 05. 替换空格.cpp
// 剑指 Offer 05. 替换空格
// https://leetcode.cn/problems/ti-huan-kong-ge-lcof/

#include <iostream>
#include <string>

std::string replaceSpace(std::string str) {
    std::string result;
    
    for (char ch : str) {
        if (ch == ' ') {
            result += "%20";
        } else {
            result += ch;
        }
    }
    
    return result;
}

int main() {
    std::string str = "We are happy.";
    
    std::string result = replaceSpace(str);
    
    std::cout << result << std::endl;
    
    return 0;
}
